src/analysis/seminorm.lean

```diff
@@ -201,9 +347,11 @@ variables [has_scalar R ℝ] [has_scalar R ℝ≥0] [is_scalar_tower R ℝ≥0 
 
 /-- Composition of a seminorm with a linear map is a seminorm. -/
 def comp (p : seminorm 𝕜 F) (f : E →ₗ[𝕜] F) : seminorm 𝕜 E :=
-{ to_fun := λ x, p(f x),
-  smul' := λ _ _, (congr_arg p (f.map_smul _ _)).trans (p.smul _ _),
-  triangle' := λ _ _, eq.trans_le (congr_arg p (f.map_add _ _)) (p.triangle _ _) }
+{ to_fun    := λ x, p (f x),
+  nonneg'   := λ x, p.nonneg _,
+  map_zero' := by rw [f.map_zero, p.map_zero],
+  smul'     := λ _ _, (congr_arg p (f.map_smul _ _)).trans (p.smul _ _),
+  add_le'   := λ _ _, by apply eq.trans_le (congr_arg p (f.map_add _ _)) (p.add_le _ _) }
```

Comment:
Can you define this for `add_monoid_seminorm` too?

Reply:
I added this.